app/src/prod/java/com/andela/art/login/LoginActivity.java

```diff
@@ -0,0 +1,212 @@
+                    }
+                });
+
+    }
+
+    /**
+     * Sign in and redirect.
+     */
+    public void signIn() {
+        // Show the dialog as we are now signing in.
+        mConnectionProgressDialog.show();
+        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                GoogleSignInAccount account = task.getResult(ApiException.class);
+                String emailDomain = account.getEmail().split("@")[1];
+
+                // filter out Andela email addresses
+                if ("andela.com".equals(emailDomain)) {
+                    Toast.makeText(this, "Andela email", Toast.LENGTH_SHORT).show();
+
+                } else {
+
+                    // filter out only specific GMail addresses assigned to the guards
+                    if (isAllowedNonAndelaEmail(account.getEmail())) {
+                        Toast.makeText(this, "Allowed non Andela email",
+                                Toast.LENGTH_SHORT).show();
+                    } else {
+                        mGoogleSignInClient.signOut();
+                        throw new ApiException(new Status(UNAUTHORIZED_CODE));
+                    }
+                }
+
+                firebasewithGoogleAuth(account);
+
+            } catch (ApiException e) {
+                // Hide the progress dialog if its showing.
+                mConnectionProgressDialog.dismiss();
+                // Google Sign In failed, update UI appropriately
+                Log.w(TAG, "Google sign in failed", e);
+                if (e.getStatusCode() == UNAUTHORIZED_CODE) {
+                    Toast.makeText(LoginActivity.this,
+                            "Please use your Andela email address",
+                            Toast.LENGTH_LONG).show();
+                } else {
+                    Toast.makeText(LoginActivity.this,
+                            "Authorization Failed: Ensure you have selected a Google account.",
+                            Toast.LENGTH_LONG).show();
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Check to see if a given non Andela GMail address is among the ones allowed to login.
+     *
+     * @param email Email address for the user
+     * @return boolean true if an email is allowed, false if the email is not allowed
+     */
+    public boolean isAllowedNonAndelaEmail(String email) {
+        return allowedEmailAddresses.contains(email);
```

Comment:
@zmwas It's possible that the login completes before the list is populated. Could you add a null/empty check?